Decred Wallet/AppDelegate.swift

```diff
@@ -20,6 +20,8 @@ import Crashlytics
 protocol AppLifeCycleDelegate {
     func applicationEnteredForegroundFromSuspendedState(_ lastActiveTime: Date)
     func networkChanged(_ connection: Reachability.Connection)
+    func applicationWillTerminate()
+    func applicationWillEnterBackground()
```

Comment:
Re-order these methods in the following order
`applicationEnteredForegroundFromSuspendedState`
`applicationWillEnterBackground`
`applicationWillTerminate`
`networkChanged`

The implementations in `Syncer.swift` should also follow same order.